object-manipulation-3/main.js

```diff
@@ -0,0 +1,116 @@
+console.log('Lodash is loaded:', typeof _ !== 'undefined');
+
+var deck = [];
+var ranks = ['Ace', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'Jack', 'Queen', 'King'];
+var suits = ['Spades', 'Clubs', 'Diamonds', 'Hearts'];
+var benScore = 0;
+var joeScore = 0;
+var tomScore = 0;
+var billScore = 0;
+
+var players = [
+  {
+    name: 'Ben',
+    hand: []
+  },
+  {
+    name: 'Joe',
+    hand: []
+  },
+  {
+    name: 'Tom',
+    hand: []
+  },
+  {
+    name: 'Bill',
+    hand: []
+  }
+];
+
+function createDeck(ranks, suits) {
+  for (var i = 0; i < ranks.length; i++) {
+    for (var j = 0; j < suits.length; j++) {
+      deck.push({
+        rank: ranks[i],
+        suit: suits[j]
+      });
+    }
+  }
+}
+createDeck(ranks, suits);
+
+function shuffleDeck(deck) {
+  return _.shuffle(deck);
+}
+var shuffledDeck = shuffleDeck(deck);
+
+function dealTwoCards() {
+  for (var i = 0; i < players.length; i++) {
+    for (var j = 0; j < 2; j++) {
+      players[i].hand.push(shuffledDeck[j]);
+      shuffledDeck.splice(0, 2);
+    }
+  }
+}
+dealTwoCards();
+
+function getScores(players) {
+  for (var i = 0; i < players.length; i++) {
+    for (var j = 0; j < players[i].hand.length; j++) {
+      var playerCards = players[i].hand[j].rank;
+
+      if (players[i].name === 'Ben') {
+        if (playerCards === 'Ace') {
+          benScore += 11;
+        } else if (playerCards === 'Jack' || playerCards === 'Queen' || playerCards === 'King') {
+          benScore += 10;
+        } else {
+          benScore += Number(playerCards);
+        }
+      }
+      if (players[i].name === 'Joe') {
+        if (playerCards === 'Ace') {
+          joeScore += 11;
+        } else if (playerCards === 'Jack' || playerCards === 'Queen' || playerCards === 'King') {
+          joeScore += 10;
+        } else {
+          joeScore += Number(playerCards);
+        }
+      }
+      if (players[i].name === 'Tom') {
+        if (playerCards === 'Ace') {
+          tomScore += 11;
+        } else if (playerCards === 'Jack' || playerCards === 'Queen' || playerCards === 'King') {
+          tomScore += 10;
+        } else {
+          tomScore += Number(playerCards);
+        }
+      }
+      if (players[i].name === 'Bill') {
+        if (playerCards === 'Ace') {
+          billScore += 11;
+        } else if (playerCards === 'Jack' || playerCards === 'Queen' || playerCards === 'King') {
+          billScore += 10;
+        } else {
+          billScore += Number(playerCards);
+        }
```

Comment:
We should try to rework this somehow, since everything is the same, except the player's name. Maybe if we had an array of scores instead of separate variables, we could use our `i` variable.

```js
var playerScores = [0, 0, 0, 0];

// start loops

playerScores[i] += ??? // depending on rank of the card
```

Reply:
Or maybe instead, we can add a `.score` property to each player.